### PR TITLE
Add channel capability for SmartBFT

### PR DESCRIPTION
**Pull request by andrew-coleman**
First part of adding channel capability for SmartBFT

Signed-off-by: andrew-coleman <andrew_coleman@uk.ibm.com>

### CODE REVIEW

**Comment by yacovm** [comment]
Why SmartBFT? We should call it V3. We shouldn't be aware of consensus intricacies.

**Review by yacovm — CHANGES_REQUESTED** [review]

File: common/capabilities/channel.go [hunk 1]
```diff
@@ -61,6 +66,8 @@ func (cp *ChannelProvider) Type() string {
 func (cp *ChannelProvider) HasCapability(capability string) bool {
 	switch capability {
 	// Add new capability names here
+	case Channel_SmartBFT:
```
[the remark this answers is not in the document]
Comment (andrew-coleman) [reply common/capabilities/channel.go:69]: Hmm, ok, but I figured we were working on a fork that would be merged into the v3 development branch

[the remark this answers is not in the document]
Comment (yacovm) [reply common/capabilities/channel.go]: we are, in the fork we can ignore the capability, but I would like to merge the capability addition into the official fabric. 
in the fork, in the meanwhile, we just build the required components. In any case, the actual BFT orderer doesn't have to honor the capability at all. What needs to honor it, is the stuff we put in fabric such as the new block signature verification, etc.

**Comment by yacovm** [comment]
what about [configtx.yaml changes](https://github.com/hyperledger/fabric/commit/a1d687f5709b72c28cd25656013b2b48fdac2f54#diff-c87a9e869434572fe8b9da3259702851c380585b561916df2654ee9f5750fc8dR100)?

**Review by yacovm — APPROVED** [review]
Looks good, thanks.

**Pull request merged**